[ews-build.webkit.org] Use UAT S3 bucket on UAT EWS instance

JonWBedard opened the pull request:
#### 4d11473e60a2a083c8e0be6b5f088c004e3f900f
<pre>
[ews-build.webkit.org] Use UAT S3 bucket on UAT EWS instance
<a href="https://bugs.webkit.org/show_bug.cgi?id=268663">https://bugs.webkit.org/show_bug.cgi?id=268663</a>
<a href="https://rdar.apple.com/122207451">rdar://122207451</a>

Reviewed by Aakash Jain.

* Tools/CISupport/Shared/generate-s3-url: Include instance suffix in S3 bucket.
* Tools/CISupport/build-webkit-org/steps.py:
(DownloadBuiltProduct): Parameterize S3 bucket.
* Tools/CISupport/ews-build/steps.py:
(SetCommitQueueMinusFlagOnPatch.run): Allow for multiple production hostnames.
(BlockPullRequest.run): Ditto.
(RemoveLabelsFromPullRequest.doStepIf): Ditto.
(LeaveComment.doStepIf): Ditto.
(CompileWebKit.evaluateCommand): Upload to S3 on the UAT instance.
(UploadFileToS3.doStepIf): Do step on the UAT instance.
(GenerateS3URL.doStepIf): Ditto.
(TransferToS3.doStepIf): Ditto.
(DownloadBuiltProduct): Parameterize S3 bucket.
(DownloadBuiltProduct.start): Download via S3 on UAT instance.
(PushCommitToWebKitRepo.doStepIf): Allow for multiple production hostnames.
(PushPullRequestBranch.doStepIf): Ditto.
(UpdatePullRequest.doStepIf): Ditto.
* Tools/CISupport/ews-build/steps_unittest.py: Handle multiple production hostnames.

Canonical link: <a href="https://commits.webkit.org/274105@main">https://commits.webkit.org/274105@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5b94c3106466bb45ec11c80982ea878c7318d7b1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37904 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16812 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40166 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40447 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/33726 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/39234 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19507 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14112 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38476 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/14171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12361 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12326 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/33888 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41719 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/34397 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34370 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/38186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/12903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/10545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36364 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/37949 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/14479 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/13336 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4918 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13931 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-administrator commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -53,9 +53,11 @@
 S3_RESULTS_URL = f'https://ews-build{custom_suffix}.s3-us-west-2.amazonaws.com/'
 CURRENT_HOSTNAME = socket.gethostname().strip()
 EWS_BUILD_HOSTNAME = 'ews-build.webkit.org'
+EWS_BUILD_UAT_HOSTNAME = 'ews-build.webkit-uat.org'
```
Add EWS_BUILD_DEV_HOSTNAME too. Or should you just keep 
EWS_BUILD_HOSTNAME=f'ews-build{custom_suffix}' ?

Note the hostname on the k8s pods are as followed, so this may break in the current production environment running on a VM.
PROD: ews-build
UAT: ews-build-uat
DEV: ews-build-dev

We are limited and couldn't add a domain to the hostname, and the "." character.

JonWBedard replied:
Probably should change these to `EWS_PRODUCTION_HOSTNAMES` and `EWS_UAT_HOSTNAMES`, does that sound like a good plan? cc @lingcherd

webkit-administrator commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -53,9 +53,11 @@
 BUG_SERVER_URL = 'https://bugs.webkit.org/'
```
Looks like there are several other production host URLs hard coded. Might want to check if they should instead be going to UAT or DEV.

aj062 reviewed (approved):
r+ after resolving Ling's concerns.

lingcherd commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -53,9 +53,11 @@
 CURRENT_HOSTNAME = socket.gethostname().strip()
-EWS_BUILD_HOSTNAME = 'ews-build.webkit.org'
+EWS_BUILD_HOSTNAMES = ['ews-build.webkit.org', 'ews-build']
+EWS_BUILD_UAT_HOSTNAMES = ['ews-build.webkit-uat.org', 'ews-build-uat']
```
Please add ews-build.webkit-dev.org and ews-build-dev to this list. You might want to change the name of this variable to "nonprod" or something. Otherwise, create another DEV list of hostnames and modify the conditions used in other places accordingly.

JonWBedard committed:
[ews-build.webkit.org] Use UAT S3 bucket on UAT EWS instance
https://bugs.webkit.org/show_bug.cgi?id=268663
rdar://122207451

Reviewed by Aakash Jain.

* Tools/CISupport/Shared/generate-s3-url: Include instance suffix in S3 bucket.
* Tools/CISupport/build-webkit-org/steps.py:
(DownloadBuiltProduct): Parameterize S3 bucket.
* Tools/CISupport/ews-build/steps.py:
(SetCommitQueueMinusFlagOnPatch.run): Allow for multiple production hostnames.
(BlockPullRequest.run): Ditto.
(RemoveLabelsFromPullRequest.doStepIf): Ditto.
(LeaveComment.doStepIf): Ditto.
(CompileWebKit.evaluateCommand): Upload to S3 on the UAT instance.
(UploadFileToS3.doStepIf): Do step on the UAT instance.
(GenerateS3URL.doStepIf): Ditto.
(TransferToS3.doStepIf): Ditto.
(DownloadBuiltProduct): Parameterize S3 bucket.
(DownloadBuiltProduct.start): Download via S3 on UAT instance.
(PushCommitToWebKitRepo.doStepIf): Allow for multiple production hostnames.
(PushPullRequestBranch.doStepIf): Ditto.
(UpdatePullRequest.doStepIf): Ditto.
* Tools/CISupport/ews-build/steps_unittest.py: Handle multiple production hostnames.

Canonical link: https://commits.webkit.org/274105@main
